packages/amesos2/src/Amesos2_KLU2_def.hpp

```diff
@@ -304,44 +236,70 @@ KLU2<Matrix,Vector>::solve_impl(
   const global_size_type ld_rhs = this->root_ ? X->getGlobalLength() : 0;
   const size_t nrhs = X->getGlobalNumVectors();
 
+  {                             // Get values from RHS B
+#ifdef HAVE_AMESOS2_TIMERS
+    Teuchos::TimeMonitor mvConvTimer(this->timers_.vecConvTime_);
+    Teuchos::TimeMonitor redistTimer( this->timers_.vecRedistTime_ );
+#endif
+    if ( is_contiguous_ == true ) {
+      Util::get_1d_copy_helper_kokkos_view<MultiVecAdapter<Vector>,
+        host_solve_array_t>::do_get(B, bValues_,
+            as<size_t>(ld_rhs),
+            ROOTED, this->rowIndexBase_);
+    }
+    else {
+      Util::get_1d_copy_helper_kokkos_view<MultiVecAdapter<Vector>,
+        host_solve_array_t>::do_get(B, bValues_,
+            as<size_t>(ld_rhs),
+            CONTIGUOUS_AND_ROOTED, this->rowIndexBase_);
+    }
+
+    // see Amesos2_Tacho_def.hpp for an explanation of why we 'get' X
```

Comment:
Just checking that this is intended to refer back to the Tacho code, and not copy/paste

Reply:
Correct. I didn't want to duplicate large comments on this. I have an issue open #7158 to address these and will do all the solvers together. I have some ideas how to handle this and then will clean up all the comments together.